sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/multi/MultiServiceEventClient.java

```diff
@@ -0,0 +1,402 @@
+                        serviceName,
+                        response.result,
+                        response.offerResources.stream()
+                                .collect(Collectors.summingInt(or -> or.getResources().size())),
+                        response.offerResources.size(),
+                        response.offerResources.size() == 1 ? "" : "s");
+                switch (response.result) {
+                case FAILED:
+                    // We should be able to safely skip this service and proceed to the next one. For this round,
+                    // the service just won't have anything added to unexpectedResources. We play it safe by telling
+                    // upstream to do a short decline for the unprocessed resources.
+                    anyFailedClients = true;
+                    for (OfferResources unexpectedInOffer : response.offerResources) {
+                        getEntry(unexpectedResources, unexpectedInOffer.getOffer())
+                                .addAll(unexpectedInOffer.getResources());
+                    }
+                    break;
+                case PROCESSED:
+                    for (OfferResources unexpectedInOffer : response.offerResources) {
+                        getEntry(unexpectedResources, unexpectedInOffer.getOffer())
+                                .addAll(unexpectedInOffer.getResources());
+                    }
+                    break;
+                }
+            }
+        }
+
+        // Return the combined listing of unexpected resources across all services:
+        return anyFailedClients
+                ? UnexpectedResourcesResponse.failed(unexpectedResources.values())
+                : UnexpectedResourcesResponse.processed(unexpectedResources.values());
+    }
+
+    /**
+     * Maps the provided status to the service that owns its task, then queries that service with the status.
+     *
+     * <p>This is an optimization which avoids querying services about task statuses that don't relate to them.
+     * <p>In addition to reducing unnecessary queries, this also improves isolation between services. They only see
+     * task statuses which relate to them.
+     */
+    @Override
+    public StatusResponse status(Protos.TaskStatus status) {
+        Optional<AbstractScheduler> service = multiServiceManager.getMatchingService(status);
+        if (!service.isPresent()) {
+            // Unrecognized service. Status for old task?
+            LOGGER.info("Received status for unknown task {}: {}",
+                    status.getTaskId().getValue(), TextFormat.shortDebugString(status));
+            return StatusResponse.unknownTask();
+        }
+        LOGGER.info("Received status for task {}: {}", status.getTaskId().getValue(), status.getState());
+        return service.get().status(status);
+    }
+
+    /**
+     * Returns a set of queues-specific endpoints to be served by the scheduler. This effectively overrides the
```

Comment:
We should not include references to the word `queues`.

Reply:
Done